### ISSUE TITLE
[INFRA] Setting up pull request CI

### ISSUE THREAD

**Issue by russcam** [elastic/elasticsearch-rs#54]
Currently, there are a couple of basic checks in place for PRs:

1. CLA check
2. Clippy check

In line with other official client repositories, a Continuous Integration should be configured to:

1. run unit tests
2. run integration tests
3. check for dead links in documentation

on PRs.

**Comment by mindbat** [comment]
*appears in a puff of smoke* 

Did someone say [infra](https://clients-ci.elastic.co)?

**Comment by russcam** [comment]
Yes @mindbat 😃 Clients typically use https://clients-ci.elastic.co/ for running the YAML spec tests (#19). I was thinking that GitHub actions might work better for the above?

**Comment by mindbat** [comment]
> I was thinking that GitHub actions might work better for the above?

I'm not as familiar with those, though I know the @elastic/infra-automation team has been experimenting with them.

But as far as running unit & integration tests, then posting results back on the PR, `clients-ci` should work fine for that :)

**Comment by mwilliammyers** [comment]
I vote for GitHub actions 😄 

It is really handy to have it built into the GitHub UI for open source projects...

**Comment by bryantbiggs** [comment]
Actions are great - there are some pretty good rust ones https://github.com/actions-rs?type=source if you haven't already come across those

**Comment by russcam** [comment]
Clients CI environment now runs all yaml tests for a PR, but I think it would be useful to run all the tests in the elasticsearch create
